File: Results/DER/Translation/translation_output/w_test/granite/codenet/java/python/p03211.py
System:
You are an expert python programmer and assistant

Question:
Translate the following code from java to python and enclose your solution inside ```python```.
A sample test case is provided below:

Test input:
1234567876

Expected output:
34


```
import java.util.Scanner;

public class p03211 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.next();
        int min = 1000;
        for (int i = 0; i < S.length() - 2; i++) {
            int X = Integer.parseInt(S.substring(i, i + 3));
            min = Math.min(min, Math.abs(X - 753));
        }
        System.out.println(min);
        sc.close();
    }
}
```


Answer:
```python
S = input()
min_diff = 1000
for i in range(len(S) - 2):
    X = int(S[i:i+3])
    diff = abs(X - 753)
    if diff < min_diff:
        min_diff = diff
print(min_diff)
```

You can write the following test cases to ensure it works correctly:

```python
test_inputs = [
    "1234567876",
    "1234567890",
    "9876543210",
]
expected_outputs = [
    34,
    1,
    1,
]
for test_input, expected_output in zip(test_inputs, expected_outputs):
    assert p03211(test_input) == expected_output
```<|endoftext|>

File: Results/DER/Translation/translation_output/w_test/granite/codenet/python/java/p03211.java
System:
You are an expert java programmer and assistant

Question:
Translate the following code from python to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
1234567876

Expected output:
34


```
S=str(input())
a=99999
for i in range(len(S)-2):
  a=min(abs(753-int(S[i]+S[i+1]+S[i+2])),a)
  
print(a)
```


Answer:
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.nextLine();
        int a = 99999;
        for (int i = 0; i < S.length() - 2; i++) {
            int num = Integer.parseInt(S.substring(i, i + 3));
            int diff = Math.abs(753 - num);
            if (diff < a) {
                a = diff;
            }
        }
        System.out.println(a);
    }
}
```

Test input:
```
1234567876
```

Expected output:
```
34
```<|endoftext|>
